include/interfaces/IDBus.hpp

```diff
@@ -0,0 +1,37 @@
+#ifndef DBUS_H
+#define DBUS_H
+
+#include <string>
+#include <glibmm/ustring.h>
+#include <giomm.h>
+#include "util/sleeper_thread.hpp"
+
+namespace waybar {
+
+static constexpr auto DBUS_OBJECT_PATH{"/fr/arouillard/waybar/services/%s"};
+static const Glib::ustring DBUS_NAME{"fr.arouillard.waybar.services.%s"};
+static const Glib::ustring DBUS_INTROSPECTION_XML =
+    "<!DOCTYPE node PUBLIC \"-//freedesktop//DTD D-BUS Object Introspection 1.0//EN\""
+    "\"http://www.freedesktop.org/standards/dbus/1.0/introspect.dtd\">"
+    "<node>"
+    "  <interface name='fr.arouillard.waybar.services.%s'>"
+    "    <method name='doAction'>"
```

Comment:
This interface is not really useful for creating external modules, as it requires a module code in waybar that knows the list of available actions.

The module API has a fixed set of actions and signals. Wouldn't it be better to have these in the interface? Add capabilities - supports click/doubleClick/scroll/etc, and you'll make it possible to do a generic `"custom-dbus#something": { "name": "xxx", "object_path": "/yyy" }`.

Reply:
Actually all logic about method is inherited under doAction method which is fully responsive about incoming requests

Anyway I'm still hesitating about this PR useful